src/index.c

```diff
@@ -2517,15 +2566,31 @@ static bool is_index_extended(git_index *index)
 	return (extended > 0);
 }
 
-static int write_disk_entry(git_filebuf *file, git_index_entry *entry)
+static int write_disk_entry(git_filebuf *file, git_index_entry *entry, const char **last)
```

Comment:
last is an in/out parameter -- do you want me to change it to two parameters as you suggested in decode_varint?